### PR TITLE
Fix dataset visualization issues for behavior1k and other datasets

### PR DESCRIPTION
**Pull request by fwd4**
This fixes two issues that prevented visualization of certain datasets like lerobot/behavior1k-task*:

1. Fixed PyArrow schema field ordering mismatch
   - PyArrow's cast() requires field names in exact order
   - Added column reordering before casting in load_nested_dataset()
   - Fixes ValueError: "Target schema's field names are not matching"

2. Fixed KeyError: 'frame_index' in visualization script
   - Changed batch["frame_index"] to batch["index"]
   - The dataset uses "index" field, not "frame_index"
   - Line 135 already correctly uses batch["index"]

Tested with: lerobot/behavior1k-task0000

## Title

Short, imperative summary (e.g., "fix(robots): handle None in sensor parser"). See [CONTRIBUTING.md](../CONTRIBUTING.md) for PR conventions.

## Type / Scope

- **Type**: (Bug | Feature | Docs | Performance | Test | CI | Chore)
- **Scope**: (optional — name of module or package affected)

## Summary / Motivation

- One-paragraph description of what changes and why.
- Why this change is needed and any trade-offs or design notes.

## Related issues

- Fixes / Closes: # (if any)
- Related: # (if any)

## What changed

- Short, concrete bullets of the modifications (files/behaviour).
- Short note if this introduces breaking changes and migration steps.

## How was this tested (or how to run locally)

- Tests added: list new tests or test files.
- Manual checks / dataset runs performed.
- Instructions for the reviewer

Example:

- Ran the relevant tests:

  ```bash
  pytest -q tests/ -k <keyword>
  ```

- Reproduce with a quick example or CLI (if applicable):

  ```bash
  lerobot-train --some.option=true
  ```

## Checklist (required before merge)

- [ ] Linting/formatting run (`pre-commit run -a`)
- [ ] All tests pass locally (`pytest`)
- [ ] Documentation updated
- [ ] CI is green

## Reviewer notes

- Anything the reviewer should focus on (performance, edge-cases, specific files) or general notes.
- Anyone in the community is free to review the PR.

### CODE REVIEW

File: src/lerobot/scripts/lerobot_dataset_viz.py [hunk 1]
```diff
@@ -133,7 +133,7 @@ def visualize_dataset(
     for batch in tqdm.tqdm(dataloader, total=len(dataloader)):
         # iterate over the batch
         for i in range(len(batch["index"])):
-            rr.set_time("frame_index", sequence=batch["frame_index"][i].item())
+            rr.set_time("frame_index", sequence=batch["index"][i].item())
```
Comment (CarolinePascal) [inline src/lerobot/scripts/lerobot_dataset_viz.py:136]: To stay compatible with the expected `frame_index` behavior, the `index` must start at 0 at the beginning of each episode. The simplest approach is to store the initial value `batch["index"][0]` from the first batch, then subtract that value from `batch["index"][i]` for all subsequent entries.

File: src/lerobot/datasets/utils.py [hunk 2]
```diff
@@ -131,6 +131,9 @@ def load_nested_dataset(
         table = arrow_dataset.to_table(filter=filter_expr)
 
         if features is not None:
+            # Reorder columns to match schema field order before casting
+            schema_field_names = features.arrow_schema.names
+            table = table.select(schema_field_names)
```
Comment (CarolinePascal) [inline src/lerobot/datasets/utils.py:136]: This should now be solved on `main` !

**Comment by CarolinePascal** [comment]
Hi @fwd4,

Because of a short deadline, your PR was superseded by #3068. Thank you for your contribution !

Best,

Caroline.
